serverlessworkflow/modules/ROOT/pages/eventing/working-with-callbacks.adoc

```diff
@@ -1 +1,122 @@
-//= Working with callbacks
\ No newline at end of file
+Working with Callbacks
+======================
+:compat-mode!:
+// Metadata:
+:description: Working with callbacks
+:keywords: kogito, workflow, serverless, callback, event
+
+
+This guide discuss the link:{spec_doc_url}#Callback-State[Callback State]. This is a state that performs an action and waits for an event produced as a result of that action to resume the workflow. So it is suitable to perform __fire&wait-for-result__ operations. The action will normally be an external service invocation. This service is expected to be async.
+
+From the flow perspective, async means that the control will be returned to the caller almost immediately, without waiting for the actual operation to be finished. Once it ends, a link:{cloud_events_url}[Cloud Event] will be published to resume the flow.
+
+For the flow to know this published event is the one it is waiting for, the external service developer might include the Kogito process instance id in the event header or, preferably, use the xref:event-correlation-with-workflows[Serverless Workflow correlation mechanism]. This guide will focus on the former mechanism, which is based on the fact that every Kogito Workflow instance has an automatically generated and unique identifier.
+
+== Error callback example 
+
+The workflow in the link:{kogito_sw_examples_url}/serverless-workflow-callback-quarkus[Serverless Workflow Callback Quarkus] example illustrates the callback feature. Once started (this application does not have an initial model), it publishes an event of type `resume` and waits for an event of `wait` type.
+
+The type `resume` event is consumed by a listener that simulates the behavior of an external service. Consequently, as an external service will do, once the actions associated with that event reception have finished, this listener publishes a new event of type `wait`. Once this event is received, the workflow moves to the next state and successfully ends. 
```

Comment:
maybe we can explicitly say the type is the cloud event `type` attribute, right?